api/app.js

```diff
@@ -0,0 +1,18 @@
+const express = require('express');
+const cors = require('cors');
+
+// middleware // 
+const server = express();
+server.use(cors());
+server.use(express.json());
+
+const postRoutes = require('./controllers/posts')
+
+server.use('/posts', postRoutes)
+
+const port = process.env.PORT || 3000;
+
+// Root route
+//server.get('/', (req, res) => res.send('Hello, world!')) - might not be needed?
```

Comment:
To fi x this you should have a server listening to requests here. You have a port but it is unused. 

so here app.listen(port,()=> {console.log("...port no"... ))})

no need to export this file as it si your main entry point actually here you import your server functionality